feat: add checkConditionalExpressions to `no-unmodified-loop-condition` (#21175)
eslint/eslint: docs/src/rules/no-unmodified-loop-condition.md

## Options

This rule has an object option:

> * `checkConditionalExpressions`: When `true`, a ternary expression does not form a group when the rule determines whether its references are modified. This does not change how binary expressions are handled. The default is `false`, which checks the result of the entire ternary expression.

Reviewer: I'm not sure if this description is clear to everybody. You could make it easier to understand by splitting the cases as it's done for other rules. For example (feel free to adjust the wording):

```suggestion
* `"checkConditionalExpressions": false` (default) checks the result of an entire ternary expression
* `"checkConditionalExpressions": true` each branch in a ternary expression is checked independently
```

Author: Done 👍

Examples of **incorrect** code for this rule with the `{ checkConditionalExpressions: true }` option:

::: incorrect

```js

> /* eslint no-unmodified-loop-condition: ["error", { checkConditionalExpressions: true }] */

Reviewer: Options in ESLint rule docs are usually double-quoted even if they are valid identifiers. Just a stylistic convention.

```suggestion
* `"checkConditionalExpressions"`: When `true`, a ternary expression does not form a group when the rule determines whether its references are modified. This does not change how binary expressions are handled. The default is `false`, which checks the result of the entire ternary expression.

Examples of **incorrect** code for this rule with the `{ "checkConditionalExpressions": true }` option:

::: incorrect

```js
/* eslint no-unmodified-loop-condition: ["error", { "checkConditionalExpressions": true }] */
```
